tests/Unit/Options/Test_Options.cpp

```diff
@@ -147,11 +150,71 @@ SPECTRE_TEST_CASE("Unit.Options.NamedSimple.invalid", "[Unit][Options]") {
   opts.get<NamedSimple>();
 }
 
+namespace {
+struct Group {
+  static constexpr OptionString help = {"Group halp"};
+};
+
+struct GroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Tag halp"};
+  using group = Group;
+};
+
+struct OuterGroup {
+  static constexpr OptionString help = {"Outer group halp"};
+};
+
+struct InnerGroup {
+  static constexpr OptionString help = {"Inner group halp"};
+  using group = OuterGroup;
+};
+
+struct InnerGroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Inner tag halp"};
+  using group = InnerGroup;
+};
+
+struct OuterGroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Outer tag halp"};
+  using group = OuterGroup;
+};
+
+void test_options_grouped() {
+  {
+    INFO("Option groups");
+    Options<tmpl::list<GroupedTag, Simple>> opts("Overall help text");
+    opts.parse(
+        "Group:\n"
+        "  GroupedTag: 3\n"
+        "Simple: 2");
+    CHECK(opts.get<GroupedTag>() == 3);
+    CHECK(opts.get<Simple>() == 2);
+  }
+  {
+    INFO("Nested option groups");
+    Options<tmpl::list<InnerGroupedTag, OuterGroupedTag, Simple>> opts(
+        "Overall help text");
+    opts.parse(
+        "OuterGroup:\n"
+        "  InnerGroup:\n"
+        "    InnerGroupedTag: 3\n"
+        "  OuterGroupedTag: 1\n"
+        "Simple: 2");
```

Comment:
It would be nice if conversion errors displayed what group the option is in (in case of duplicate names in different groups).  Maybe in the parse backtrace.  This is not critical since it correctly gives the line and column numbers.

Reply:
Added the backtrace

Reply:
I personally like this backtrace. I think that makes it clear what is happening :)